source/code/plugins/change_tracking_runner.rb

```diff
@@ -0,0 +1,105 @@
+require "rexml/document"
+require "cgi"
+require 'logger'
+require 'digest'
+require 'json'
+
+require_relative 'changetracking_lib'
+require_relative 'oms_common'
+
+
+class ChangeTrackingRunner 
+	CHANGE_TRACKING_FILE = "/etc/opt/omi/conf/omsconfig/configuration/ChangeTrackingInventory.xml"
+	CHANGE_TRACKING_STATE_FILE = "/var/opt/microsoft/omsagent/state/ChangeTrackingHash.conf"
+	LAST_UPLOAD_HASH = "PREV_HASH"
+	LAST_UPLOAD_TIME = "LAST_UPLOAD_TIME"
+
+	@log =  Logger.new(STDERR) #nil
+	@log.formatter = proc do |severity, time, progname, msg|
+	        "#{severity} #{msg}\n"
+	end
+
+	def self.transform_and_wrap()
+		if File.exist?(CHANGE_TRACKING_FILE)
+			@log.debug ("Found the change tracking inventory file.")
+
+			# Get the parameters ready.
+			time = Time.now
+			force_send_run_interval_hours = 24
+			force_send_run_interval = force_send_run_interval_hours.to_i * 3600
+			@hostname = OMS::Common.get_hostname or "Unknown host"
+
+			# Read the inventory XML.
+			file = File.open(CHANGE_TRACKING_FILE, "rb")
+			xml_string = file.read; nil # To top the output to show up on STDOUT.
+
+			#Transform the XML to HashMap
+			transformed_hash_map = ChangeTracking.transform(xml_string, @hostname, time, @log)
+			@log.debug("transformed_hash_map size - #{transformed_hash_map["packages"].size}, #{transformed_hash_map["services"].size}, #{transformed_hash_map["fileInventories"].size}")
+
+			output = ChangeTracking.wrap(time, @hostname, transformed_hash_map)
+			@log.debug("output size - #{output.to_s}")
```

Comment:
I'l convert the`@log.debug("output size - #{output.to_s}")` to only look for lengths of dataitems.